src/main/java/reposense/git/GitRemote.java

```diff
@@ -0,0 +1,43 @@
+package reposense.git;
+
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+import reposense.system.CommandRunner;
+
+/**
+ * Contains git remote related functionality.
+ * Git remote is responsible for getting remote repository information.
+ */
+public class GitRemote {
+
+    /**
+     * Extracts remote repository information. Returns a map
+     * with keys of the form REMOTE_NAME(fetch) or REMOTE_NAME(push).
+     */
+    public static Map<String, String> getRemotes(String repoRoot) {
+        Map<String, String> remotes = new HashMap<>();
+        String result;
+        try {
+            result = CommandRunner.runCommand(Paths.get(repoRoot), "git remote -v");
+        } catch (RuntimeException re) {
+            return remotes;
```

Comment:
Do you think some kind of logged message is necessary here? Something like 'no remote repository was found'

Reply:
I initially did have a log message here actually. I removed it because this catch will generally only hit when the directory isn't even a git repository. It seemed a bit strange to be logging that here, since the report generation will have to fail eventually anyway.

Reply:
Because right now, if the command is run against a purely local repository as opposed to a non-git repository, the outcomes are the same and there is no way to distinguish. I thought it might be helpful to add in the event that down the line, some relative pathing issue results in the command being run in the wrong directory altogether. Or a fully remote repository being mistaken for a non-git directory.

Reply:
Yes, `git remote -v` will still succeed in a purely local git repository. The closest log message to the truth that i can think of is something like: `Unable to run git remote command in directory: %s`. Do you have any suggestions for improving this?

Reply:
I think that is good enough. I think this should be mostly for defensive programming since I don't expect that line to ever be reached given proper execution. But we never know 🤷